### PR TITLE
Hp fix ensure lb

### PR DESCRIPTION
**Pull request by houpeng80**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/cloud-controller-manager/main.go [hunk 1]
```diff
@@ -1,12 +1,9 @@
 /*
  Copyright 2022 The Kubernetes Authors.
-
```
Comment (chengxiangdong) [inline cmd/cloud-controller-manager/main.go:3]: keep the empty line

File: cmd/cloud-controller-manager/main.go [hunk 2]
```diff
@@ -1,12 +1,9 @@
...
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
-
```
Comment (chengxiangdong) [inline cmd/cloud-controller-manager/main.go:7]: ditto

File: cmd/cloud-controller-manager/main.go [hunk 3]
```diff
@@ -1,12 +1,9 @@
...
      http://www.apache.org/licenses/LICENSE-2.0
-
```
Comment (chengxiangdong) [inline cmd/cloud-controller-manager/main.go:9]: ditto

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 4]
```diff
@@ -19,15 +19,13 @@ package huaweicloud
...
-	"strconv"
-	"strings"
-
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
 	cloudprovider "k8s.io/cloud-provider"
 	"k8s.io/klog/v2"
+	"strconv"
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: It's not a good import style, usually we move all imports in a single declaration, group stdlib imports, and group imports.
IDEA Setting: Settings -> Editor -> Code Style -> Go

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 5]
```diff
@@ -131,8 +130,8 @@ func (l *LB) getLoadBalancerInstance(ctx context.Context, clusterName string, se
 		return nil, err
 	}
 
-	name := l.GetLoadBalancerName(ctx, clusterName, service)
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: why remove these?

File: pkg/cloudprovider/huaweicloud/loadbalancer.go [hunk 6]
```diff
@@ -434,6 +433,9 @@ func (l *LB) getOrCreatePool(loadbalancer *services.LoadBalancer,
 	pool, err := ensureOpts.lbServices.GetPool(loadbalancer.ID, listener.ID)
 	if err != nil && common.IsNotFound(err) {
 		pool, err = l.createPool(poolName, listener, ensureOpts)
+		if err != nil {
```
Comment (chengxiangdong) [inline pkg/cloudprovider/huaweicloud/loadbalancer.go]: here's a better way to write it

File: cmd/cloud-controller-manager/main.go (hunk 2 above)
Comment (houpeng80) [reply cmd/cloud-controller-manager/main.go:7]: OK

File: cmd/cloud-controller-manager/main.go (hunk 3 above)
Comment (houpeng80) [reply cmd/cloud-controller-manager/main.go:9]: OK

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 4 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: OK

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 6 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: ok

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 4 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: It has been fixed.

File: pkg/cloudprovider/huaweicloud/loadbalancer.go (hunk 5 above)
Comment (houpeng80) [reply pkg/cloudprovider/huaweicloud/loadbalancer.go]: I have change the code back to this.

**Pull request merged**